### ISSUE TITLE
2.0 Release Timeline?

### ISSUE THREAD

**Issue by brycepg** [PyCQA/astroid#540]
How for in the future do you expect the 2.0 release to be? 

I could work on unblocking 2.0 instead of backporting fixes if the release is near enough. What do you think @PCManticore?

I know PyCQA/pylint#1914 is blocking release. Anything else?

**Comment by PCManticore** [comment]
Hey Bryce, thanks for creating this issue!
I was thinking that a release around the date of Python 3.7 release would be great to have, so somewhere in June.

While initially our milestones where a bit messy and probably not very reasonable, I updated them for both pylint and astroid to contain just the minimum of what could be great to release for each milestone. For 2.0, for both pylint and astroid, I only added Python 3.7 support and various performance improvements we could do until then. Let me know if this sounds reasonable enough: https://github.com/PyCQA/pylint/milestone/2 and https://github.com/PyCQA/astroid/milestone/3.

Regarding Python 3.7, I was handling data classes locally, and I'll have a PR soon-ish with it, but there seems to be other improvements that we could support: https://docs.python.org/3.7/whatsnew/3.7.html. First and foremost though would be to make it run under Python 3.7, so having it in CI would be great for testing what doesn't work yet.

Would definitely appreciate all the help in releasing 2.0! That's probably more important right now than backporting fixes.

**Comment by hroncok** [comment]
Could you please release 2.0 in alpha or beta even before Python 3.7 is released? It would help us get things prepped for the release in Fedora. I can help, however I suppose one needs rights, etc. to do a release.

**Comment by PCManticore** [comment]
Yes, we can do that, pretty much everything should work for 3.7 so there's nothing blocking the release per se. I'll release an alpha version later this week.

**Comment by hroncok** [comment]
Thank you. Should I open another issue for pylint, or do you release in sync?

**Comment by PCManticore** [comment]
It's all in sync.

**Comment by PCManticore** [comment]
@hroncok I just released astroid==2.0.dev. I'll do the same in a bit for pylint as well. Please give it a go and let me know if you notice something breaking!

**Comment by PCManticore** [comment]
Closing this issue for now. For folks that want to help, please check the 2.0 milestone if there's anything there that might need some help. Also adding better support for Python 3.7 before the release is also a good idea, as well as testing 2.0.dev for both astroid and pylint to see if anything breaks on the latest 3.7 RC.
